File: python/intro to Python/2 - loops and conditions/program.py
if __name__ == "__main__":
    # for loop
    sum = 0

    for i in range(10, 101, 5):  # don't need parenthesis around condition
        sum += i

    print(f"10 + 15 + 20 + ... + 100 = {sum}")

    # while loop; lets find the first capital letter
    word = "abcEFgHIj"
    i = 0
    while (i < len(word)) and not word[i].isupper():
        i += 1  # no i++ in Python

    if i != len(word):  # Python doesn't null-terminate strings, so we do this
        print(f"the first capital letter of \"{word}\" is '{word[i]}'.")
    else:
        print(f"{word} contains no capital letters.")

    # if, else if, else
    choice = 'b'  # NOTE: Python doesn't care if you use single/double quotes!

    if choice == 'a':
        print("choice 1")
    elif choice == 'b':  # NOTE: Python uses `elif`, not `else if`
        print("choice 2")
    else:
        print("other choice")

File: python/intro to Python/2 - loops and conditions/program.cpp
#include <iostream>
#include <cstring>

using namespace std;

int main()
{
    // for loop
    int sum = 0;

    for (int i = 10; i < 101; i += 5) {
        sum += i;
    }

    cout << "10 + 15 + 20 + ... + 100 = " << sum << endl;

    // while loop; lets find the first capital letter
    const char * word = "abcEFgHIj";
    int i = 0;
    while (i < strlen(word) && !isupper(word[i])) {
        ++i;
    }

    if (word[i] != '\0')
        cout << "the first capital letter of \"" << word << "\" is '"
             << word[i] << "'." << endl;
    else
        cout << word << " contains no capital letters." << endl;


    // if, else if, else
    char choice = 'b';

    if (choice == 'a')
        cout << "choice 1" << endl;
    else if (choice == 'b')
        cout << "choice 2" << endl;
    else
        cout << "other choice" << endl;

    return 0;
}